homeassistant/components/tellduslive/__init__.py

```diff
@@ -71,33 +71,31 @@
 
     hass.data[DATA_CONFIG_ENTRY_LOCK] = asyncio.Lock()
     hass.data[CONFIG_ENTRY_IS_SETUP] = set()
-
-    client = TelldusLiveClient(hass, entry, session)
-    hass.data[DOMAIN] = client
-    await async_add_hubs(hass, client, entry.entry_id)
-    hass.async_create_task(client.update())
-
-    interval = timedelta(seconds=entry.data[KEY_SCAN_INTERVAL])
-    _LOGGER.debug('Update interval %s', interval)
-    hass.data[INTERVAL_TRACKER] = async_track_time_interval(
-        hass, client.update, interval)
+    hass.data[NEW_CLIENT_TASK] = hass.loop.create_task(
+        async_new_client(hass, session, entry))
 
     return True
 
 
-async def async_add_hubs(hass, client, entry_id):
+async def async_new_client(hass, session, entry):
     """Add the hubs associated with the current client to device_registry."""
+    interval = entry.data[KEY_SCAN_INTERVAL]
+    _LOGGER.debug('Update interval %s seconds.', interval)
+    client = TelldusLiveClient(hass, entry, session, interval)
+    hass.data[DOMAIN] = client
     dev_reg = await hass.helpers.device_registry.async_get_registry()
     for hub in await client.async_get_hubs():
         _LOGGER.debug("Connected hub %s", hub['name'])
         dev_reg.async_get_or_create(
-            config_entry_id=entry_id,
+            config_entry_id=entry.entry_id,
             identifiers={(DOMAIN, hub['id'])},
             manufacturer='Telldus',
             name=hub['name'],
             model=hub['type'],
             sw_version=hub['version'],
         )
+    await client.update()
+    return True
```

Comment:
When are we checking this return value?

Reply:
I read [this](https://docs.python.org/3/library/asyncio-task.html#asyncio.Task.done) as the task should return a value.

Reply:
> A Task is done when the wrapped coroutine either **returned a value**, raised an exception, or the Task was cancelled.

Reply:
@MartinHjelmare it would be great if this could make 0.87.0..

Reply:
I think those docs are not clear enough if that really is the case. I think they just mean that the coroutine should have returned. We don't `return True` in other parts of home assistant where we use `task.done`. Testing also indicates that we don't need to return explicitly.

```py
In [25]: async def no_return():
    ...:     print('no_return running')
    ...:     await asyncio.sleep(2)
    ...:     

In [26]: async def with_return():
    ...:     print('with_return running')
    ...:     await asyncio.sleep(2)
    ...:     return True
    ...:     

In [27]: async def main(loop, coro):
    ...:     task = loop.create_task(coro)
    ...: 
    ...:     # Wait for 1 second
    ...:     await asyncio.sleep(1)
    ...: 
    ...:     if task.done():
    ...:         print('task is done:', task)
    ...:     else:
    ...:         print('task is not done:', task)
    ...:         print('waiting for task')
    ...:         await task
    ...:         print('task is done == ', task.done())
    ...:     print('exiting main')
    ...:     

In [28]: loop = asyncio.get_event_loop()

In [29]: loop.run_until_complete(main(loop, no_return()))
no_return running
task is not done: <Task pending coro=<no_return() running at <ipython-input-25-4bec5e62923f>:3> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7fe6ec6f0438>()]>>
waiting for task
task is done ==  True
exiting main

In [30]: loop.run_until_complete(main(loop, with_return()))
with_return running
task is not done: <Task pending coro=<with_return() running at <ipython-input-26-8ee68c4dd335>:3> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7fe6ec6f0738>()]>>
waiting for task
task is done ==  True
exiting main

In [31]: 
```